addons/website/models/website.py

```diff
@@ -1331,7 +1331,7 @@ def _get_canonical_url_localized(self, lang, canonical_params):
             path = urls.url_quote_plus(request.httprequest.path, safe='/')
         lang_path = ('/' + lang.url_code) if lang != self.default_lang_id else ''
         canonical_query_string = '?%s' % urls.url_encode(canonical_params) if canonical_params else ''
-        return self.get_base_url() + lang_path + path + canonical_query_string
+        return self.get_base_url() + ((lang_path + path).rstrip('/') or '/') + canonical_query_string
```

Comment:
What do you think about a specific patch for the homepage?
This generic patch will prevent `@route()` defined with trailing slashes to ever be considered as canonical as the canonical will always be the one without trailing slash.
I know we have removed all such definition in our codebase in 15.0 with https://github.com/odoo/odoo/commit/c1ae086cdb596115235e90473b6d263a2914050e and https://github.com/odoo/enterprise/commit/b6d6608b7a9d6d811c675ea92e03768e4394e8bc but I think it would be better to keep supporting those for the custos right?
(And maybe we could even add a test for that in `test_website` with a trailing slash defined route to ensure the canonical is built with the trailing slash (and so reachable)
```suggestion
        localized_path = lang_path + path
        if lang_path and path == '/':
            # We want `/fr_BE` not `/fr_BE/` for correct canonical on homepage
            localized_path = lang_path
        return self.get_base_url() + localized_path + canonical_query_string

```